EIPS/eip-3664.md

```diff
@@ -0,0 +1,314 @@
+
+    /**
+     * @dev Remove the attribute from NFT.
+     */
+    function remove(uint256 _nftId, uint256 _attrId) external;
+
+    /**
+     * @dev attribute upgrade level for nft.
+     */
+    function upgradeLevel(uint256 _nftId, uint256 _attrId, uint8 _level) external;
+
+    /**
+     * @dev attribute upgrade sub level for nft.
+     */
+    function upgradeSubLevel(uint256 _nftId, uint256 _attrId, uint8 _subLevel) external;
+
+    event UpgradableAttributeCreated(string name, uint256 id);
+    event UpgradableAttributeAttached(uint256 nftId, uint256 attrId);
+    event UpgradableAttributeRemoved(uint256 nftId, uint256 attrId);
+    event AttributeLevelUpgraded(uint256 nftId, uint256 attrId, uint8 level);
+    event AttributeSubLevelUpgraded(uint256 nftId, uint256 attrId, uint8 subLevel);
+}
+
+
+```
+
+### IEvolutiveAttribute
+
+```
+pragma solidity ^0.8.0;
+
+interface IEvolutiveAttribute {
+    /**
+     * @dev Returns the name of the attribute .
+     */
+    function name(uint256 _attrId) external view returns (string memory);
+
+    /**
+     * @dev Returns the description of the attribute.
+     */
+    function description(uint256 _attrId) external view returns (string memory);
+
+    /**
+     * @dev Returns the attributes of the nft.
+     */
+    function getNFTAttrs(uint256 _nftId) external view returns (uint256[] memory);
+
+    /**
+     * @dev Returns the max level of the attribute.
+     */
+    function maxLevel(uint256 _attrId) external view returns (uint8);
+
+    /**
+     * @dev Create new attribute.
+     */
+    function create(
+        uint256 _id,
+        string memory _name,
+        string memory _description,
+        uint8 _level,
+        uint8[] memory _probabilities,
+        uint256[] memory _block_intervals
+    ) external;
+
+    /**
+     * @dev Attach the attribute to NFT.
+     */
+    function attach(uint256 _nftId, uint256 _attrId) external;
+
+    /**
+     * @dev Remove the attribute from NFT.
+     */
+    function remove(uint256 _nftId, uint256 _attrId) external;
+
+    /**
+     * @dev Evolutive the attribute of the NFT.
+     */
+    function evolutive(uint256 _nftId, uint256 _attrId, uint8 _level) external;
+
+    /**
+     * @dev Evolution has the probability to damage the NFT, if it is damaged, it can be repaired.
+     */
+    function repair(uint256 _nftId, uint256 _attrId) external;
+
+    event EvolutiveAttributeCreated(string name, uint256 id);
+    event EvolutiveAttributeAttached(uint256 nftId, uint256 attrId);
+    event EvolutiveAttributeRemoved(uint256 nftId, uint256 attrId);
+    event AttributeEvolutive(uint256 nftId, uint256 attrId, uint8 level);
+    event AttributeRepaired(uint256 nftId, uint256 attrId);
+}
+
+
+```
+
+## Test Cases
+[gNFT/test at main · DRepublic-io/gNFT (github.com)](https://github.com/DRepublic-io/gNFT/tree/main/test)
+
+##  Implementation
+[DRepublic-io/gNFT (github.com)](https://github.com/DRepublic-io/gNFT/)
+
+## Additional References
```

Comment:
Please remove this section, it isn't a valid section per EIP-1.

Reply:
Thank you lightclient, edited! Did you mean Additional References section? Removed, thanks!